test/parallel/test-net-connect-local-error.js

```diff
@@ -3,25 +3,40 @@ const common = require('../common');
 const assert = require('assert');
 const net = require('net');
 
-const server = net.createServer();
-server.listen(0);
-const port = server.address().port;
-const client = net.connect({
-  port: port + 1,
-  localPort: port,
-  localAddress: common.localhostIPv4
-});
+function getUnassignedPort(cb) {
+  const server = net.createServer();
+  server.listen({port: 0}, common.mustCall(() => {
+    // When the server is closed this port will no longer be assigned
+    const unassignedPort = server.address().port;
+    server.close(common.mustCall(() => {
+      cb(unassignedPort);
+    }));
+  }));
+}
 
-client.on('error', common.mustCall(function onError(err) {
-  assert.strictEqual(
-    err.localPort,
-    port,
-    `${err.localPort} !== ${port} in ${err}`
-  );
-  assert.strictEqual(
-    err.localAddress,
-    common.localhostIPv4,
-    `${err.localAddress} !== ${common.localhostIPv4} in ${err}`
-  );
+getUnassignedPort(common.mustCall((unassignedPort) => {
+  assert(unassignedPort);
```

Comment:
@refack Fixed! Thanks! I've also moved the `getUnassignedPort` call closer to where the value is actually used.